### PR TITLE
Intbig

### PR DESCRIPTION
**Pull request by Icermli**
BigInt unit test

### CODE REVIEW

**Review by faddat — COMMENTED** [review]
So if I understand correctly, we're basically adding another type here, `bigint`?

File: src/test/scala/vsys/blockchain/contract/DataEntrySpecification.scala [hunk 1]
```diff
@@ -40,19 +73,88 @@ class DataEntrySpecification extends PropSpec with PropertyChecks with Generator
...
+    val byteArray6 = Array.fill[Byte](1 + 4)(4)
+    val byteArray7 = Array[Byte](5, 0, 0)
+    val byteArray8 = Array.fill[Byte](1 + 26)(6)
+    val byteArray9 = Array.fill[Byte](1 + 26)(7)
+    val byteArray10 = Array.fill[Byte](1 + 8)(9)
+    val byteArray11 = Array[Byte](10, 1)
+    val byteArray12 = Array[Byte](11, 0, 0)
+    val byteArray13 = Array[Byte](12, 118, 97, 108, 117, 101, 49)
+    val byteArray14 = Array[Byte](13, 0, 0)
+    val byteArray15 = Array[Byte](14, 0, 0)
```
Comment (ncying) [inline src/test/scala/vsys/blockchain/contract/DataEntrySpecification.scala:142]: just confused the representation of empty byte array in BigInteger~ @zl730

File: src/test/scala/vsys/blockchain/contract/DataEntrySpecification.scala [hunk 2]
```diff
@@ -40,19 +73,88 @@ class DataEntrySpecification extends PropSpec with PropertyChecks with Generator
...
+    DataEntry.fromBytes(byteArray6).map(_.dataType) should be (Right(DataType.Int32))
+    DataEntry.fromBytes(byteArray7).map(_.dataType) should be (Right(DataType.ShortText))
+    DataEntry.fromBytes(byteArray8).map(_.dataType) should be (Left(InvalidDataEntry))
+    DataEntry.fromBytes(byteArray9).map(_.dataType) should be (Left(InvalidDataEntry))
+    DataEntry.fromBytes(byteArray10).map(_.dataType) should be (Right(DataType.Timestamp))
+    DataEntry.fromBytes(byteArray11).map(_.dataType) should be (Right(DataType.Boolean))
+    DataEntry.fromBytes(byteArray12).map(_.dataType) should be (Right(DataType.ShortBytes))
+    DataEntry.fromBytes(byteArray13).map(_.dataType) should be (Left(InvalidDataEntry))
+    DataEntry.fromBytes(byteArray14).map(_.dataType) should be (Right(DataType.OpcBlock))
+    DataEntry.fromBytes(byteArray15).map(_.dataType) should be (Right(DataType.BigInteger))
```
Comment (ncying) [inline src/test/scala/vsys/blockchain/contract/DataEntrySpecification.scala:157]: @Icermli empty byte array is special here, may add more general cases

File: src/test/scala/vsys/blockchain/state/opcdiffs/BasicOpcDiffTest.scala [hunk 3]
```diff
@@ -132,6 +209,20 @@ class BasicOpcDiffTest extends PropSpec with PropertyChecks with GeneratorDriven
...
+      Shorts.toByteArray(BigInt(2).toByteArray.length.toShort) ++ BigInt(2).toByteArray, DataType.BigInteger))))
+    BasicOpcDiff.operation(
+      DataEntry(Shorts.toByteArray(BigInt(0).toByteArray.length.toShort) ++ BigInt(0).toByteArray, DataType.BigInteger),
+      DataEntry(Shorts.toByteArray(BigInt(0).toByteArray.length.toShort) ++ BigInt(0).toByteArray, DataType.BigInteger),
+      Seq.empty, 1, BasicOpcDiff.maximum) should be (Left(ContractLocalVariableIndexOutOfRange))
+    BasicOpcDiff.operation(
+      DataEntry(Shorts.toByteArray(BigInt(0).toByteArray.length.toShort) ++ BigInt(0).toByteArray, DataType.BigInteger),
+      DataEntry(Longs.toByteArray(1), DataType.Timestamp),
+      Seq.empty, 0, BasicOpcDiff.maximum) should be (Left(ContractDataTypeMismatch))
+
```
Comment (ncying) [inline src/test/scala/vsys/blockchain/state/opcdiffs/BasicOpcDiffTest.scala:225]: @Icermli an sqrt method is added for BigInteger DataEntry, test it too

File: src/test/scala/vsys/blockchain/contract/DataEntrySpecification.scala (hunk 2 above)
Comment (Icermli) [reply src/test/scala/vsys/blockchain/contract/DataEntrySpecification.scala:157]: Sure

File: src/test/scala/vsys/blockchain/state/opcdiffs/BasicOpcDiffTest.scala (hunk 3 above)
Comment (Icermli) [reply src/test/scala/vsys/blockchain/state/opcdiffs/BasicOpcDiffTest.scala:225]: ok

File: src/test/scala/vsys/blockchain/contract/DataEntrySpecification.scala (hunk 1 above)
Comment (zl730) [reply src/test/scala/vsys/blockchain/contract/DataEntrySpecification.scala:142]: `java.lang.NumberFormatException: Zero length BigInteger` would be caused

**Review by ncying — APPROVED** [review]

**Comment by ncying** [comment]
May add more general basic operation test cases such as non-empty byte array later.
